dvc/utils/__init__.py

```diff
@@ -292,14 +292,22 @@ def fix_env(env=None):
     return env
 
 
-def convert_to_unicode(data):
-    if isinstance(data, builtin_str):
-        return str(data)
-    if isinstance(data, dict):
-        return dict(map(convert_to_unicode, data.items()))
-    if isinstance(data, (list, tuple)):
-        return type(data)(map(convert_to_unicode, data))
-    return data
+if builtin_str is str:
+
+    def convert_to_unicode(data):
```

Comment:
How does `data` depend on whether or not `builtin_str` is a str or not?

Reply:
The function converts all `builtin_str` to `str` (imported from `utils.compat`) in a nested structure, which is a noop if they are the same, basically in Python 3.